docs/lang/articles/misc/profiler.md

```diff
@@ -4,78 +4,125 @@ sidebar_position: 4
 
 # Profiler
 
-Taichi's profiler can help you analyze the run-time cost of your
-program. There are two profiling systems in Taichi: `KernelProfiler` and
-`ScopedProfiler`.
+## Overview
 
-`KernelProfiler` is used to analyze the performance of user kernels.
+High-performance numerical computation is one of the key application domains of Taichi.
+We provide a suite of profilers to quantify the performance of Taichi programs, help analyze where the bottleneck occurs,
+and thus facilitate users optimizing their code. These profilers collect both hardware and Taichi-related information
+and can also be used as performance debugging tools for developers.
 
-While `ScopedProfiler` is used by Taichi developers to analyze the
-performance of the compiler itself.
+The follows are profiling tools Taichi provides now:
+- `ScopedProfiler` is used to analyze the performance of the Taichi JIT compiler (host).
+- `KernelProfiler` shows the performance of Taichi kernels (device), with detailed low-level performance metrics (such as memory bandwidth consumption) in its advanced mode.
 
-## KernelProfiler
+## ScopedProfiler
 
-1.  `KernelProfiler` records the costs of Taichi kernels on devices. To
-    enable this profiler, set `kernel_profiler=True` in `ti.init`.
-2.  Call `ti.print_kernel_profile_info()` to show the kernel profiling
-    result. For example:
+`ScopedProfiler` tracks the time spent on **host tasks** such as JIT compilation.
 
-```python {3,13}
+1. This profiler is automatically on.
+2. Call `ti.print_profile_info()` to display results in a hierarchical format.
+
+For example:
+
+```python
 import taichi as ti
 
-ti.init(ti.cpu, kernel_profiler=True)
+ti.init(arch=ti.cpu)
 var = ti.field(ti.f32, shape=1)
 
-
 @ti.kernel
 def compute():
     var[0] = 1.0
-
+    print("Setting var[0] =", var[0])
 
 compute()
-ti.print_kernel_profile_info()
-```
-
-The outputs would be:
-
-```
-[ 22.73%] jit_evaluator_0_kernel_0_serial             min   0.001 ms   avg   0.001 ms   max   0.001 ms   total   0.000 s [      1x]
-[  0.00%] jit_evaluator_1_kernel_1_serial             min   0.000 ms   avg   0.000 ms   max   0.000 ms   total   0.000 s [      1x]
-[ 77.27%] compute_c4_0_kernel_2_serial                min   0.004 ms   avg   0.004 ms   max   0.004 ms   total   0.000 s [      1x]
+ti.print_profile_info()
 ```
 
 :::note
-Currently the result of `KernelProfiler` could be incorrect on OpenGL
-backend due to its lack of support for `ti.sync()`.
+`ScopedProfiler` is a C++ class in the core of Taichi. It is not exposed to Python users.
```

Comment:
nit: better not discourage Python users. Maybe for Python profiling one can use...?

Reply:
How about we briefly note ` `ScopedProfiler` is a C++ class in Taichi. `